lib/actions/invitation.ts

```diff
@@ -0,0 +1,411 @@
+    });
+
+    if (existingUser) {
+      const existingMember = await (
+        prisma as any
+      ).organizationMember.findUnique({
+        where: {
+          userId_organizationId: {
+            userId: existingUser.id,
+            organizationId,
+          },
+        },
+      });
+
+      if (existingMember) {
+        return { error: "User is already a member of this organization" };
+      }
+    }
+
+    // Check for existing pending invitation
+    const existingInvitation = await (
+      prisma as any
+    ).organizationInvitation.findFirst({
+      where: {
+        email,
+        organizationId,
+        acceptedAt: null,
+      },
+    });
+
+    if (existingInvitation) {
+      return { error: "An invitation has already been sent to this email" };
+    }
+  }
+
+  const token = crypto.randomUUID();
+  const expiresAt = expiresInDays ? addDays(new Date(), expiresInDays) : null;
+
+  try {
+    const invitation = await (prisma as any).organizationInvitation.create({
+      data: {
+        email,
+        role,
+        token,
+        expiresAt,
+        organizationId,
+        invitedById: session.user.id,
+      },
+      include: {
+        organization: true,
+        invitedBy: true,
+      },
+    });
+
+    // Send email invitation if email is provided
+    if (email) {
+      const baseUrl = process.env.VERCEL_URL
+        ? `https://${process.env.VERCEL_URL}`
+        : process.env.NEXTAUTH_URL || "http://app.localhost:3000";
+      const inviteUrl = `${baseUrl}/accept-invite?token=${token}`;
+
+      const emailContent = React.createElement(InvitationEmail, {
+        inviteUrl,
+        organizationName: invitation.organization.name || "an organization",
+        inviterName: invitation.invitedBy.name || invitation.invitedBy.email,
+        role: role,
+        expiresAt,
+      });
+
+      try {
+        await sendEmail({
+          to: email,
+          from: "Mailer",
+          subject: `Invitation to join ${invitation.organization.name || "organization"}`,
+          react: emailContent,
+          previewText: `You've been invited to join ${invitation.organization.name}`,
+        });
+      } catch (emailError) {
+        console.error("Error sending invitation email:", emailError);
+        // Don't fail the invitation creation if email fails
+      }
+    }
+
+    revalidatePath(`/organization/${organizationId}/settings/members`);
+
+    return { success: true, invitation };
+  } catch (error) {
+    console.error("Error creating invitation:", error);
+    return { error: "Failed to create invitation" };
+  }
+}
+
+// Accept an invitation
+export async function acceptInvitation(token: string) {
+  const session = await getSession();
+  if (!session?.user?.id) {
+    return { error: "Not authenticated" };
+  }
+
+  const invitation: any = await (
+    prisma as any
+  ).organizationInvitation.findUnique({
+    where: { token },
+    include: { organization: true },
+  });
+
+  if (!invitation) {
```

Comment:
I would recommend replacing specific error language with generic error language for security reasons. In this setup, attackers could poke the system and get information out, especially the message "This invitation was sent to ${invitation.email}. Please sign in or create an account with that email address."

At minimum, I would remove the email address from the message.

As written, an attacker could iterate through invitations and get registered emails out, as well as more info. I recommend generic language so attackers can't know if a token exists, has been used, is expired, or belongs to an email address.